rundeckapp/grails-app/controllers/rundeck/interceptors/AuthorizationInterceptor.groovy

```diff
@@ -1,27 +1,38 @@
 package rundeck.interceptors
 
+import com.codahale.metrics.Counter
 import com.dtolabs.client.utils.Constants
 import com.dtolabs.rundeck.app.api.ApiVersions
 import org.rundeck.app.access.InterceptorHelper
+import org.springframework.beans.factory.InitializingBean
 
 import javax.servlet.http.HttpServletResponse
 
-class AuthorizationInterceptor {
+class AuthorizationInterceptor implements InitializingBean {
+
+    def metricService
 
     int order = HIGHEST_PRECEDENCE + 50
 
     InterceptorHelper interceptorHelper
 
+    Counter apiAuthorizationSuccess
+    Counter apiAuthorizationFailure
+
     AuthorizationInterceptor() {
         matchAll()
     }
 
     boolean before() {
-        if(interceptorHelper.matchesAllowedAsset(controllerName, request)) return true
+        if(interceptorHelper.matchesAllowedAsset(controllerName, request)) {
+            apiAuthorizationSuccess.inc()
+            return true
+        }
         if(request.apiVersionStatusNotReady){
             response.status = HttpServletResponse.SC_SERVICE_UNAVAILABLE
             return false
         }else if (request.invalidApiAuthentication) {
```

Comment:
Just out of curiosity, the `return true` after this `if` doesn't mean anything regarding those counters, right?

Reply:
That's correct